### PR TITLE
adding FakeGeneva

### PR DESCRIPTION
**Pull request by 1ucian0**
This PRs adds FakeGeneva based on `ibm_geneva`.

### CODE REVIEW

**Review by HuangJunye — CHANGES_REQUESTED** [review]
Looks great! The only thing I want to comment is the ordering. Shouldn't Geneva go before Guadalupe?

**Comment by 1ucian0** [comment]
> Looks great! The only thing I want to comment is the ordering. Shouldn't Geneva go before Guadalupe?

Thanks! Fixed in 6cf67f5c2384d112422ee13075f89b7f9d4075a3

**Review by HuangJunye — DISMISSED** [review]
Thanks for fixing the order. LGTM!

**Comment by nonhermitian** [comment]
Before accepting these kind of PRs, one should check that the snapshot is actually a good one.  Here some of the 2Q gates have errors of 1, eg.:

```
"qubits": [16, 14], "gate": "cx", "parameters": [{"date": "2022-07-11T17:50:35.854518+02:00", "name": "gate_error", "unit": "", "value": 1}
```

**Comment by 1ucian0** [comment]
good catch. I will update `update_fake_backends.py` to warning about it.

**Comment by HuangJunye** [comment]
In addition to #8331, should we also add a test for checking the gate error and other qubit properties?

**Comment by 1ucian0** [comment]
> Before accepting these kind of PRs, one should check that the snapshot is actually a good one. Here some of the 2Q gates have errors of 1, eg.:

Like `ibm_washington`, exploratory systems seems to have this kind of issues. I will keep an eye checking for new updates. I still think there is value in having it as a fakebackend, even with errors=1.

**Comment by nonhermitian** [comment]
So the difference is that Geneva is a new 27Q system that probably still has some kinks to work out.  You are grabbing a snapshot of it when there are 7 CX gates (13%) with faulty calibrations.  If anything you should either wait a bit (there is no rush), or grab a cal that is at least as good as one can get given the device history.  For example, the following timestamps will give you calibrations with only 2 bad CX gates:

```
2022-06-25 06:06:47.365299
2022-06-24 06:06:48.326497
2022-06-23 06:06:49.151597
```

With respect to Washington, the first system of its family, it has 284 CX gates in the coupling map, and it is possible to get calibrations for that system with only 4 bad cx gates (2%).  So selecting a calibration that has 13% bad cx gates on a processor family that has been around for a couple of years seems a bit off to me.  Perhaps one can just wait until close to the next release date and see how things pan out before adding a snapshot?

**Comment by 1ucian0** [comment]
awesome! thanks for the tip @nonhermitian 

Updating this PR with the calibration from `2022-06-24 23:00:30+02:00`

```
ibm_geneva
==========
Gate errors == 1:
 * cx 20, 19
 * cx 16, 14
╒══════════════╤═════════════════════════════════╤═════════════════════════════════════╕
│   error == 1 │  new 2022-06-24 23:00:30+02:00  │  current 2022-07-12 14:40:53+02:00  │
╞══════════════╪═════════════════════════════════╪═════════════════════════════════════╡
│     cx error │          2/56 (3.57%)           │            7/56 (12.50%)            │
├──────────────┼─────────────────────────────────┼─────────────────────────────────────┤
│    all gates │          2/191 (1.05%)          │            7/191 (3.66%)            │
╘══════════════╧═════════════════════════════════╧═════════════════════════════════════╛
```

File: releasenotes/notes/ibm_geneva-5b1e9308dc302e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    The fake backend :class:`~FakeGeneva` was added to with the information from IBM Quantum `ibm_geneva` system.
```
Comment (1ucian0) [inline releasenotes/notes/ibm_geneva-5b1e9308dc302e2e.yaml:4]:
```suggestion
    The fake backend :class:`~FakeGeneva` was added with the information from IBM Quantum `ibm_geneva` system.
```

File: qiskit/providers/fake_provider/backends/geneva/__init__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Mock geneva backend"""
```
Comment (HuangJunye) [inline qiskit/providers/fake_provider/backends/geneva/__init__.py:13]: Similar comment to other fake backend PRs. https://github.com/Qiskit/qiskit-terra/pull/8344#discussion_r941378461

```suggestion
"""Fake Geneva device (27 qubits)."""
```

**Comment by 1ucian0** [comment]
This one should be done too @mtreinish

File: releasenotes/notes/ibm_geneva-5b1e9308dc302e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply releasenotes/notes/ibm_geneva-5b1e9308dc302e2e.yaml]: ups.. fixed.

**Review by mtreinish — APPROVED** [review]

**Pull request merged**